NC | IAM | Docs | Update Terminology Part (#8306)
noobaa/noobaa-core: docs/design/iam.md

- IAM UpdateAccessKey: AccessKeyId, Status, UserName
- IAM DeleteAccessKey: AccessKeyId, UserName
- IAM ListAccessKeys: UserName (not supported: Marker, MaxItems)

## Other
### Terminology - AWS vs NooBaa
|   | AWS | NooBaa |
|---|-----|--------|
|   | root account  | account  |
|   | IAM user  | user  |

#### Root Account / Account

> - Since we understand that NC (NooBaa Non-Containerized) deployment (a lightweight deployment of NooBaa on Linux) the term "root" is associated with Linux root permission, we decided that we will use the term "account" instead of "root account" in our use.

Reviewer: ```suggestion
- In NooBaa NC, the term "root" is associated with Linux root permission, therefore, the term "account" will be the equivalent term used for "root account".
```

Author: Changed, thanks.

>   - The account is the owner of the users that he created using the IAM API. The account owns the users he owns and manage them (can create, read, update, delete or list them).

Author: I removed the "he" and changed it to "it" since it is about accounts or entities in the system.

  - The account is the owner of the buckets that he or his IAM users created.
- In AWS root accounts are only created in the console.  

> While in NooBaa, accounts can be created using the CLI or by using the IAM API CreateUser by account that has the flag `--iam_operate_on_root_account`. An account that has this flag operates the IAM API on accounts instead of users, but it does not own them.

Reviewer: ```suggestion
While in NooBaa, accounts can be created by - 
1. NooBaa CLI `account add` command.
2. IAM API CreateUser operation. The requesting account must have the `iam_operate_on_root_account` property set to true. An account that has `iam_operate_on_root_account` property set to true, will operate on accounts instead of users when calling the IAM API, although it does not own them.
```

Author: Changed, thanks.

> - In NooBaa account is identified:  

Reviewer: In NooBaa, an account is identified by:

Author: Changed, thanks.

  - by name  - in the CLI we pass the account name. The account name is unique within all the accounts (you cannot create a new account with the name of an existing account).
  - by access key - in S3 API and IAM API the request is signed with the requesting account credentials.

#### Identity
- In general, we manage identities - currently accounts and users - but in the future, we might support roles, groups, etc.).

#### IAM User / User

> - IAM users aren't separate accounts; they're individual users within an account (for a single person or application). IAM users and their access keys have long-term credentials to the system resource, they give the ability to make programmatic requests to AWS services using the API or CLI.  

Reviewer: ```suggestion
- IAM users are individual users within an account (for a single person or application), they aren't separate accounts. 
- IAM users and their access keys have long-term credentials to the system resource, they give the ability to make programmatic requests to AWS services using the API or CLI.  
```

Author: Changed, thanks.

This was partially copied from [AWS IAM Guide - Intro](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction_identity-management.html#intro-identity-users) and [AWS IAM Guide - When To Use IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/when-to-use-iam.html#security_iam_authentication-iamuser).

> - In NooBaa user is identified:

Reviewer: ```suggestion
- In NooBaa, a user is identified by:
```

Author: Changed, thanks.
